app/controllers/users/invitations_controller.rb

```diff
@@ -25,6 +26,14 @@ def validate_invited_user
     return_user_to_invite_page if user_is_invalid?
   end
 
+  def user_should_be_cleared?
+    resending_invite? || unconfirmed_user_with_no_org?
+  end
+
+  def unconfirmed_user_with_no_org?
+    invited_user_already_exists? && invited_user_not_confirmed? && invited_user_has_no_org?
```

Comment:
We are doing three separate user lookups here, thats very inefficient. We should just be doing one.

Reply:
would argue this should be done as part of a separate refactor, as it's a pre-existing issue for this class

Reply:
Cool, added to the sprint